homeassistant/components/notify_events/__init__.py

```diff
@@ -1,22 +1,63 @@
-"""The notify_events component."""
+"""Support to trigger Maker Notify.Events recipes."""
+import logging
+
 import voluptuous as vol
 
-from homeassistant.const import CONF_TOKEN, Platform
+from homeassistant.components import webhook
+from homeassistant.const import CONF_TOKEN, CONF_WEBHOOK_ID, Platform
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers import discovery
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
 from homeassistant.helpers.typing import ConfigType
 
 from .const import DOMAIN
 
+_LOGGER = logging.getLogger(__name__)
+
+PARAM_ACTION = "action"
+
 CONFIG_SCHEMA = vol.Schema(
-    {DOMAIN: vol.Schema({vol.Required(CONF_TOKEN): cv.string})}, extra=vol.ALLOW_EXTRA
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_TOKEN): cv.string,
+                vol.Optional(CONF_WEBHOOK_ID): cv.string,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
 )
 
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the notify_events component."""
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Notify.Events service component."""
+    if DOMAIN not in config:
+        return True
 
     hass.data[DOMAIN] = config[DOMAIN]
-    discovery.load_platform(hass, Platform.NOTIFY, DOMAIN, {}, config)
+
+    hass.async_create_task(
+        async_load_platform(hass, Platform.NOTIFY, DOMAIN, {}, config)
+    )
+
+    if CONF_WEBHOOK_ID in config[DOMAIN]:
+        webhook.async_register(
+            hass,
+            DOMAIN,
+            "Notify.Events",
+            config[DOMAIN][CONF_WEBHOOK_ID],
+            handle_webhook,
+        )
+
     return True
+
+
+async def handle_webhook(hass, webhook_id, request):
+    """Handle webhook callback."""
+    params = request.query
+
+    if PARAM_ACTION not in params:
+        _LOGGER.error("Received invalid data from Notify.Events. %s", params)
+        return
+
+    hass.bus.async_fire(params[PARAM_ACTION])
```

Comment:
This means we can publish *any* event. Is that wanted, or should we prefix the event with `notify_events_` or something?

Reply:
Not really sure about this one. I guess if I will force users to prefix action names (which after become events) they won't be able to use event names they already have automations for. Am I missing something? 🙂

Reply:
To give you more details: "actions" in our plugin are just basically "events" that we allow the user to trigger right from the notifications we sent to him.

Scenario: User set's up an automation which sends notification (using our plugin) that there's someone comes to door. Notification contains photo from camera and "open_door" action. In the notification that we will send to our user, that action will be shown as a button. And if user pressed the button, our service will make a callback, which will trigger the "open_door" event.

Reply:
@emontnemery, any comments? :)